src/Glpi/Console/Database/UpdateCommand.php

```diff
@@ -91,10 +92,10 @@ class UpdateCommand extends AbstractCommand implements ConfigurationCommandInter
     #[Override]
     public function getSpecificMandatoryRequirements(): array
     {
-        /** @var \DBmysql $DB */
+        /** @var DBmysql|null $DB */
         global $DB;
 
-        return [new DatabaseTablesEngine($DB)];
+        return $DB instanceof DBmysql ? [new DatabaseTablesEngine($DB)] : [];
```

Comment:
You can probably use `$this->db`.

Also, it produces an exception if the configuration file is present but DB is not reachable (`Uncaught Exception Error: mysqli object is not fully initialized in /var/www/glpi/src/DBmysql.php at line 384`).

```suggestion
        return $this->db instanceof DBmysql && $this->db->connected ? [new DatabaseTablesEngine($this->db)] : [];
```